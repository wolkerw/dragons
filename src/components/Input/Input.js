import React, { Component } from "react";
import PropTypes from "prop-types";
import "./Input.scss";
import { Form } from "react-bootstrap";

class Input extends Component {
    constructor() {
        super();
        this.state = {
            firstInteraction: true,
            valid: false,
            value: null
        };
    }

    componentWillMount() {
        if (this.state.value !== this.props.value)
            this.setState({ value: this.props.value });
    }

    render() {
        const { id, placeholder, isRequired, name } = this.props;

        //this.setState({ value: value });

        const handleChange = event => {
            this.setState({ value: event.target.value });

            const value = event.target.value.length > 0 ? true : false;
            if (isRequired) {
                event.target.value = event.target.value.replace(/[\s]/g, "");
                this.setState({
                    valid: value
                });
                this.props.validateCallback(event.target.value, name, value);
            } else {
                this.props.validateCallback(event.target.value, name, true);
            }
        };

        const onBlur = () => {
            this.setState({ firstInteraction: false });
        };

        return (
            <div className="input" onBlur={onBlur}>
                <Form.Control
                    type="text"
                    id={id}
                    autoComplete="off"
                    name={name}
                    value={this.state.value}
                    placeholder={placeholder}
                    className={
                        "form-control" +
                        (!isRequired ||
                        (this.props.firstSubmit &&
                            (this.state.firstInteraction || this.state.valid))
                            ? ""
                            : " is-invalid")
                    }
                    onChange={handleChange}
                />
            </div>
        );
    }
}

Input.propTypes = {
    //id: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    firstSubmit: PropTypes.bool,
    isRequired: PropTypes.bool
};

export default Input;
