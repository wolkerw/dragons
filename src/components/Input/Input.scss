.input {
    input {
        padding: 0 54px;
        margin: 0 auto;
    }
}
