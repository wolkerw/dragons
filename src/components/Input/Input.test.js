import React from "react";
import Input from "./Input";
import { IntlProvider } from "react-intl";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import messages from "../../assets/i18n";

configure({ adapter: new Adapter() });

const country = "pt";
const language = "pt-BR";
const placeholder = messages[language].login;
const callback = () => {
    console.log("TESTE");
};
const firstInteraction = true;
const loginIsRequired = true;

describe("Input Component", () => {
    const component = (
        <IntlProvider
            locale={country}
            key={country}
            messages={messages[{ country }]}
            defaultLocale={country}
        >
            <Input
                placeholder={placeholder}
                validateCallback={callback}
                firstSubmit={firstInteraction}
                isRequired={loginIsRequired}
            />
        </IntlProvider>
    );

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("generated an input with the correct props", () => {
        const wrapper = mount(component);
        const props = wrapper.children().props();
        expect(props.placeholder).toEqual(placeholder);
        expect(props.firstSubmit).toEqual(firstInteraction);
        expect(props.validateCallback).toEqual(callback);
        expect(props.isRequired).toEqual(loginIsRequired);
    });
});
