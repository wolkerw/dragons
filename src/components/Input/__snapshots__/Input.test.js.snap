// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches the snapshot 1`] = `
<div
  className="input"
  onBlur={[Function]}
>
  <input
    autoComplete="off"
    className="form-control form-control"
    onChange={[Function]}
    placeholder="Login"
    type="text"
  />
</div>
`;
